elements/lisk-p2p/src/p2p.ts

```diff
@@ -785,30 +693,30 @@ export class P2P extends EventEmitter {
 		if (this._isActive) {
 			throw new Error('Cannot start the node because it is already active');
 		}
-		await this._startPeerServer();
 
 		// Fetch status of all the seed peers and then start the discovery
-		const seedPeerInfos = await this._fetchSeedPeerStatus(
-			this._config.seedPeers,
-		);
+		const seedPeerInfos = this._config.seedPeers;
 
 		// Add seed's peerinfos in tried peer as we already tried them to fetch status
 		seedPeerInfos.forEach(seedInfo => {
 			const peerId = constructPeerIdFromPeerInfo(seedInfo);
-			if (!this._triedPeers.has(peerId)) {
-				this._triedPeers.set(peerId, seedInfo);
+			if (!this._newPeers.has(peerId)) {
+				this._newPeers.set(peerId, seedInfo);
 			}
 		});
 
-		await this._startDiscovery();
-		this._startPopulator();
+		await this._startPeerServer();
+
+		if (this._isActive) {
```

Comment:
this if seems strange.
It's throwing in the above if it's active, and it's checking _isActive

Reply:
Ah yep it does look strange, I will add a comment. It's needed because there is an edge case which can cause `_isActive` to be false if the P2P library is stopped before it fully started (while in the middle of _startPeerServer) because promises resolve on the next tick. There could be a millisecond gap between when `this._isActive = true` is set inside `_startPeerServer` and when this line is hit and it's possible that `_isActive` could have been set to `false` during this time.